dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/EurodatSafeDepositBoxInitialization.kt

```diff
@@ -0,0 +1,59 @@
+package org.dataland.datalandexternalstorage.services
+
+import jakarta.annotation.PostConstruct
+import org.dataland.datalandeurodatclient.openApiClient.api.SafeDepositDatabaseResourceApi
+import org.dataland.datalandeurodatclient.openApiClient.model.SafeDepositDatabaseRequest
+import org.dataland.datalandeurodatclient.openApiClient.model.SafeDepositDatabaseResponse
+import org.dataland.datalandexternalstorage.utils.EurodatDataStoreUtils.retryWrapperMethod
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+/**
+ * Simple implementation of the initialization of the eurodat minabo safedepositbox
+ * @param safeDepositDatabaseResourceClient the service to create the safe deposit box used to store private data
+ * on eurodat
+ */
+@Component
+class EurodatSafeDepositBoxInitialization(
+    @Autowired var safeDepositDatabaseResourceClient: SafeDepositDatabaseResourceApi,
+    @Value("\${dataland.eurodatclient.app-name}")
+    private val eurodatAppName: String,
+    @Value("\${dataland.eurodatclient.initialize-safe-deposit-box}")
+    private val initializeSafeDepositBox: Boolean,
+
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Tries to create a safe deposit box in EuroDaT for storage of Dataland data a pre-defined number of times and
+     * then throws a final exception after the retries are used up.
+     */
+    @PostConstruct
+    fun createSafeDepositBox() {
+        if (initializeSafeDepositBox) {
+            logger.info("Checking if safe deposit box exits. If not creating safe deposit box")
+            retryWrapperMethod("createSafeDepositBox") {
+                isSafeDepositBoxAvailable()
+            }
+        }
+    }
+
+    @Suppress("TooGenericExceptionThrown")
```

Comment:
Nicht notwendig, wenn du einfach eine ServiceUnavailableException oder so schmeißt. Oder irgendeine andere spezifischere.

Reply:
Wobei man ja eigentlich nicht davon ausgehen kann, dass der service unavailable ist.

Es ist dann eher der Fall, dass etwas Unerwartetes zurückgekommen ist.

Also vllt eher 

throw IllegalArgumentException("Received unexpected response when trying to create safe deposit box.");

?

Reply:
Geändert